channel.go

```diff
@@ -140,13 +129,21 @@ type Channel struct {
 // channelConnectionCommon is the list of common objects that both use
 // and can be copied directly from the channel to the connection.
 type channelConnectionCommon struct {
-	log             Logger
-	relayStats      relay.Stats
-	statsReporter   StatsReporter
-	traceReporter   TraceReporter
-	subChannels     *subChannelMap
-	timeNow         func() time.Time
-	traceSampleRate float64
+	log           Logger
+	relayStats    relay.Stats
+	statsReporter StatsReporter
+	tracer        opentracing.Tracer
+	subChannels   *subChannelMap
+	timeNow       func() time.Time
+}
+
+// Tracer returns the OpenTracing Tracer for this channel.
+// If no tracer was provided in the configuration, returns opentracing.GlobalTracer().
+func (ccc channelConnectionCommon) Tracer() opentracing.Tracer {
+	if ccc.tracer != nil {
```

Comment:
shouldn't we only need to do this check once when creating the channel rather than everytime `Tracer` is called?